src/main/java/bob/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,119 @@
+package bob.commands;
+
+import java.io.IOException;
+
+import bob.data.task.Deadline;
+import bob.data.task.Tasklist;
+import bob.exceptions.BobEmptyTaskException;
+import bob.exceptions.BobInvalidDateAndTimeException;
+import bob.storage.Storage;
+import bob.ui.Ui;
+
+/**
+ * Adds a deadline to the Bob's tasklist.
+ */
+public class DeadlineCommand extends Command {
+    private final String input;
+
+    /**
+     * Creates a deadline command.
+     *
+     * @param input User input.
+     * @throws BobEmptyTaskException If there is no description for deadline.
+     */
+    public DeadlineCommand(String input) throws BobEmptyTaskException {
+        //Removes all whitespaces and checks if input is empty
+        if (input.replaceAll("\\s+", "").length() == 0) {
+            throw new BobEmptyTaskException();
+        }
+
+        this.input = input.startsWith(" ")
+                    ? input.substring(1)
+                    : input;
+    }
+
+    /**
+     * Formats date to be parsed.
+     *
+     * @param date Inputted date.
+     * @return Formatted date.
+     * @throws BobInvalidDateAndTimeException If the inputted date and time has invalid format.
+     */
+    private String formatDate(String date) throws BobInvalidDateAndTimeException {
+        String[] split = date.split("/");
+        // If length of date is not same as "YYYY/MM/DD" and components are not separated by "/"
+        if (date.length() != 10 || split.length != 3) {
+            throw new BobInvalidDateAndTimeException();
+        }
+        return split[0] + "-" + split[1] + "-" + split[2];
+    }
```

Comment:
You can consider using dateformat to reformat your date's presentation. Could potentially reduce errors and it is easier to catch exceptions.

Reply:
Oh okay! Ill take note of that, thanks!